src/Components/Navbar/navbar.js

```diff
@@ -34,11 +36,25 @@ const Navbar = () => {
             )}
           </span>
         </button>
-        <button className="btn-com btn-icon-singular">
-          <span className="btn-icon">
-            <i class="fa-solid fa-user"></i>
-          </span>
-        </button>
+        {token ? (
+          <Link to="/explore">
+            {" "}
+            <button className="btn-com btn-icon-singular">
+              <span className="btn-icon">
+                <i class="fa-solid fa-user"></i>
+              </span>
+            </button>
+          </Link>
+        ) : (
+          <Link to="/login">
+            {" "}
+            <button className="btn-com btn-icon-singular">
+              <span className="btn-icon">
+                <i class="fa-solid fa-user"></i>
+              </span>
+            </button>
+          </Link>
```

Comment:
only difference I see here is Like 'to' value, so you can apply a ternary to here itself instead of entire Link tag